### ISSUE TITLE
[Bug] compressed-tensors FP8 lm_head weight_scale silently dropped → degenerate repetition with unsloth/Qwen3.8-27B-NVFP4

### ISSUE THREAD

**Issue by liulifox233** [sgl-project/sglang#34895]
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [x] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Describe the bug

Serving [unsloth/Qwen3.8-27B-NVFP4](https://huggingface.co/unsloth/Qwen3.8-27B-NVFP4) (compressed-tensors mixed-precision: NVFP4 W4A4 MLP layers 0–55, FP8 W8A8 per-channel linear_attn / layers 56–63 MLP / **lm_head**) with sglang produces severely degraded output: the model repeats a single phrase forever inside reasoning (e.g. ``"need analysis there need analysis there need analysis there ..."``) from the very first token, even for a short 67-token prompt, and finishes with empty content (`finish_reason=length`). The same checkpoint serves correctly with vLLM.

## Root cause

`CompressedTensorsConfig.get_quant_method()` only dispatches `LinearBase` and `FusedMoE`. `ParallelLMHead` (a `VocabParallelEmbedding` subclass) falls through to `return None` → `UnquantizedEmbeddingMethod`, so the FP8 `lm_head.weight_scale` is never loaded:

```
Parameter lm_head.weight_scale not found in params_dict
```

The raw FP8 `lm_head.weight` (values ~±448) is then consumed as if it were BF16. The per-channel scales (mean ≈ 1.55e-4, row-to-row spread up to 12×) are never applied, so logits ranking is badly corrupted (top-10 tokens overlap only 2/10 with the correct dequantized head) → garbage/looping generation.

## Related issues/PRs

- **#28130** (PR, open): proposes exactly this `ParallelLMHead` dispatch, but only supports WNA16 (pack-quantized) heads and explicitly rejects W8A8-FP8. This model's FP8 lm_head needs the same dispatch without that restriction.
- **vllm-project/vllm#37291** (merged): vLLM fixed the identical gap and routes **all** matching schemes (including W8A8-FP8) through `CompressedTensorsLinearMethod`. I verified the same approach works in sglang: `LogitsProcessor._compute_lm_head` already routes a quantized head through `quant_method.apply()` via `should_apply_lm_head_quant_method`, so FP8 W8A8 lm_head is computed correctly on the FP8 CUTLASS path.

### Reproduction

```bash
python -m sglang.launch_server --model-path unsloth/Qwen3.8-27B-NVFP4 \
    --tp 2 --attention-backend flashinfer --kv-cache-dtype auto \
    --chunked-prefill-size 2048 --mamba-radix-cache-strategy extra_buffer_lazy \
    --reasoning-parser qwen3 --tool-call-parser qwen3_coder \
    --max-running-requests 4 --cuda-graph-max-bs 4 --mem-fraction-static 0.78
```

Request: `Which is larger, 9.11 or 9.9? Answer briefly.`

### Environment

- sglang commit 41cd5a7 (2026-08-14), TP=2
- Hardware: 2× RTX 5090 (sm120), flashinfer 0.6.17

**Comment by jahnclawdmonet** [comment]
Independently verified: this is fixed on current main by #35228 (merged 2026-08-19, four days after this report), which added `get_lm_head_scheme` dispatch for `ParallelLMHead` in `CompressedTensorsConfig.get_quant_method`. The checkpoint's `re:.*lm_head` target now resolves to the W8A8-FP8 channel scheme and `lm_head.weight_scale` loads.

A/B on one machine, same venv, same flags, single RTX PRO 6000 Blackwell (SM120, 96 GB), TP=1, `--attention-backend flashinfer`, temp 0. Only variable is the checkout:

| | pre-fix (`c7e2c08d1`, parent of #35228) | current main (`07a9de25b`) |
|---|---|---|
| `lm_head.weight_scale not found` warnings at load | 1 | 0 |
| "Which is larger, 9.11 or 9.9? Answer briefly." | degenerate repetition inside reasoning, empty content | correct: `9.9` |
| "Name the capital of Australia and one fact about it." | degenerate repetition, empty content | correct: Canberra plus a valid fact |
| finish_reason | `length` (burns max_tokens, both prompts) | `stop` (both prompts) |

Pre-fix repetition sample, same shape as reported: `...Name Australia's one city one city one city one city one city...` with `content_len 0`.

Also reproduces and resolves at TP=1, so the failure mode is independent of the TP=2 sharding in the original report.

Two notes for anyone hitting this on a release build:
- v0.5.18 (latest release at time of writing) was cut from a branch point that predates #35228, so the fix is not in any released version yet; you need main past 5375babb.
- Unrelated to this bug: `sglang[all]==0.5.18` currently does not resolve from PyPI (the `cuda-tile==1.6.0rc5` pin has no matching distribution), which blocks reproducing on the release wheel at all.

With the root cause fixed on main and verified against this exact checkpoint, this can probably be closed as completed via #35228.
